python/dask_cudf/dask_cudf/core.py

```diff
@@ -163,8 +163,10 @@ def join(self, other, **kwargs):
         return super().join(other, how=how, on=on, shuffle="tasks", **kwargs)
 
     @_dask_cudf_nvtx_annotate
-    def set_index(self, other, sorted=False, divisions=None, **kwargs):
-        if kwargs.pop("shuffle", "tasks") != "tasks":
+    def set_index(
+        self, other, sorted=False, divisions=None, shuffle="tasks", **kwargs
+    ):
+        if shuffle not in ("tasks", "explicit-comms"):
```

Comment:
Do we want to do the same pass-through in `merge` and `join` as well?

Reply:
Yeah, good call. I updated this PR to leverage the existing mechanism for activating the explicit-comms shuffle with `dask.config`, and use the context wrapper in `set_index`, `shuffle`, `sort_values`, `merge`, and `join`.